### ISSUE TITLE
harmonize `parameters` argument documentation

### ISSUE THREAD

**Issue by EmilHvitfeldt** [tidymodels/tune#831]
I think these could all be identical, would be nice to remove duplication.

<img width="823" alt="Screenshot 2024-01-23 at 12 59 32 PM" src="https://github.com/tidymodels/tune/assets/14034784/8b3e3df4-a985-4b4f-9f53-283ac28734e8">

Additionally, I think we could clarify a little bit too.

**Comment by simonpcouch** [comment]
Agreed, though they do differ in whether there's a default argument and what that default is🤔

**Comment by EmilHvitfeldt** [comment]
Yes that is true, `finalize()` and `new_iteration_results()` doesn't default to `NULL`.
